### ISSUE TITLE
multiple polygon features are picked in the 3D mode

### ISSUE THREAD

**Issue by imakihi** [TerriaJS/terriajs#6981]
I have a simple square mesh polygon layer like below. When I picked  one polygon, multiple polygons were selected. This won't happen in the 2D mode as well as when a camera look directly below in the 3D mode.
![image](https://github.com/TerriaJS/terriajs/assets/4824436/97eb9a12-6482-4bcb-bd48-b11b0b7b9fd3)

I made a short video clip to explain what I mean.
https://app.screencast.com/PZdpkBU7d9fM8

Does someone know how to avoid this behavior?

Here is a sample data.
[sample.json](https://github.com/TerriaJS/terriajs/files/13442755/sample.json)

**Comment by nf-s** [comment]
Yeah that is very strange. This only happens when using cesium primitives (instead of the new protomaps/geojson-vt renderer).

Definitely a bug here, not sure what is causing it. It might be a bug in cesium, as we let it do heavy lifting with feature picking. We will look into it 🙂

**Comment by ljowen** [comment]
I'm able to reproduce selecting multiple features getting picked using the sample.json provided, I can also reproduce it with other data sources i.e. https://www.nationalmap.gov.au/#share=s-gpC6cPOcnSVtAEo3HW1cV3ZxKK6 by zooming out and clicking near where multiple features meet.

As @nf-s  mentioned above this is handled at the cesium level so without further modifications of cesium itself we can't configure this behaviour. Ideally multiple features would be ordered by distance from the pick point, that's not currently the case.

imakihi: In your video it looks like the selected feature is quite far from the cell being clicked, is that the case or is it a latency with the screen capture?

**Comment by imakihi** [comment]
@ljowen 
Sorry for my super slow response. I haven't checked your posting.
In my case, the selected feature and clicked feature are far from each other. There was no latency with the screen capture.
Thank you.

**Comment by ljowen** [comment]
I believe this is a bug with cesiums picking when the features are clamped to ground, I've reproduced it [here](https://sandcastle.cesium.com/#c=nVTbbtswDP0VIU8JEMiOnbhJlhYb2mLAkKEFGqwY5j3IFu2olaVMkpOmRf999C1Lby/zgwWRPOTREcVUK+sISCk2Vgu+AmOYUNdGbwUHQ06Jgh05ByvKgl5+ENUffIpVWifaaSM7N4JbYGugmdHF7VFEDeygWwG71xV/1LZ+3Evr/blWDmFg4t6QPMWKkIRZWLI9mGuR3oOZk4xJC8PK5Zoa8xec0PPcFPU8wmHj1hiHxVmOTlxbEBGWGPhTCgOcOE0Kdg/ErYFstNznSJekkhWbytUi6oRVCmCc6IzYtd4JlSPI6DJfE+FIdXpSKhQsR5viFaQ5NLUpKKC51AnQmtUKSX1pOP1Tsz5bTb7TLL9Dc5uDM8dudGlQKso471catDJ+Bf3NanVxiKBSsyaCkLi3dm5j556XC+toLty6TEoLBks4UI6muvDknd6B8vwZH2dJkk5GPPAhDVk4naR+EMA0C0dhwD3Ddt50EgXZLOBR4s+mfDaFceSzMMh4Og5OZn4aAY+yyM/qepmQMKLuweGVNnyemqVReKW/1lrNiTMltBH4odiEIFa+cHieLp3EBnnfeq6lxhZpRal39Oflcnl1O6wkrWKfq2UQq6ZHSKfto9bFSvfzu0Erf6xumOIps05CpfZKa5kw8x1U2f9VAdtTOHhwc5S4u8RLxRIJ/HBYvEWQkGJMVqrUCa1If3CQgLxoDwtdL/SPW7qiVFOvUzb/D6pfCPv/5d2b0fDR1HhD6PfxO5daYY+VHDDDKJzSk+BkfJgfkrnOF05oGPnRwcVBOoZ2n/r+qM7XG/YW1u0lnHWEP4tio40jpZF9Sj0HxQYzgvWSEscDtrK1nVwL7xi64GJLBD99Z9JUjWgterJSyhvxCHHvbOFh/Bto9ajwzV9twUi2r8LWo7NlY6SULjzcvo90Tfu8yvwX) 

This issue looks related https://github.com/CesiumGS/cesium/issues/10091 .
In the meantime you can try having `forceCesiumPrimitives: false` to use the protomaps vector renderer although you would lose styling, see: https://docs.terria.io/guide/connecting-to-data/catalog-type-details/geojson/

**Comment by imakihi** [comment]
@ljowen 
Thank you for the tip and the posting. It is very good information. While I wait a parmanetn fix, I am going to test a few settings suggested.

Thank you again.
